pkg/build/node/cni.go

```diff
@@ -214,135 +213,10 @@ spec:
         securityContext:
           privileged: false
           capabilities:
-            add: ["NET_ADMIN"]
+            add: ["NET_RAW", "NET_ADMIN"]
       volumes:
         - name: cni-cfg
           hostPath:
             path: /etc/cni/net.d
 ---
-apiVersion: policy/v1beta1
-kind: PodSecurityPolicy
-metadata:
-  name: ip-masq-agent
-  annotations:
-    seccomp.security.alpha.kubernetes.io/allowedProfileNames: docker/default
-    seccomp.security.alpha.kubernetes.io/defaultProfileName: docker/default
-    apparmor.security.beta.kubernetes.io/allowedProfileNames: runtime/default
-    apparmor.security.beta.kubernetes.io/defaultProfileName: runtime/default
-spec:
-  privileged: false
-  volumes:
-    - configMap
-  readOnlyRootFilesystem: false
-  # Users and groups
-  runAsUser:
-    rule: RunAsAny
-  supplementalGroups:
-    rule: RunAsAny
-  fsGroup:
-    rule: RunAsAny
-  # Privilege Escalation
-  allowPrivilegeEscalation: false
-  defaultAllowPrivilegeEscalation: false
-  # Capabilities
-  allowedCapabilities: ['NET_RAW', 'NET_ADMIN']
-  defaultAddCapabilities: []
-  requiredDropCapabilities: []
-  # Host namespaces
-  hostPID: false
-  hostIPC: false
-  hostNetwork: true
-  # SELinux
-  seLinux:
-    rule: 'RunAsAny'
----
-kind: ClusterRole
-apiVersion: rbac.authorization.k8s.io/v1beta1
-metadata:
-  name: ip-masq-agent
-rules:
-  - apiGroups:
-      - policy
-    resources:
-      - podsecuritypolicies
-    verbs:
-      - use
-    resourceNames:
-      - ip-masq-agent
----
-kind: ClusterRoleBinding
-apiVersion: rbac.authorization.k8s.io/v1beta1
-metadata:
-  name: ip-masq-agent
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: ClusterRole
-  name: ip-masq-agent
-subjects:
-- kind: ServiceAccount
-  name: ip-masq-agent
-  namespace: kube-system
----
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-  name: ip-masq-agent
-  namespace: kube-system
----
-# ipmasq agent
-kind: ConfigMap
-apiVersion: v1
-metadata:
-  name: ip-masq-agent
-  namespace: kube-system
-  labels:
-    tier: node
-    app: ip-masq-agent
-    k8s-app: ip-masq-agent
-data:
-  config: |-
-    nonMasqueradeCIDRs:
-      - {{ .PodSubnet }}
```

Comment:
to me this is a better seperation of concerns, but we can address that in a future change

Reply:
well, per k8s network design the pod2pod traffic must not be Masquerade.
We are obtaining other nodes pod subnet to install a route to them ... and we have to route and not masquerade that traffic ... however, if we are able to get the whole PodSubnet it will allow us to separate them